.github/workflows/build.yml

```diff
@@ -0,0 +1,179 @@
+name: Build
+
+on:
+  push:
+    branches: [ develop, master ]
+  pull_request:
+    branches: [ develop ]
+  release:
+    types: [ published ]
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+
+    services:
+      postgres:
+        image: postgres:13.2
+        env:
+          POSTGRES_DB: postgres        
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_USER: postgres
+        ports:
+          - 5432:5432
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Install Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.16.x
+
+    - name: Install Node
+      uses: actions/setup-node@v2
+      with:
+        node-version: '14'
+
+    - name: Checkout
+      run: git fetch --prune --unshallow --tags
```

Comment:
I noticed this on the stash repo too...
The docs for `actions/checkout` say [you can disable the shallow checkout](https://github.com/actions/checkout/blob/main/README.md#Fetch-all-history-for-all-tags-and-branches).
So theoretically you should be able to remove this step, and instead amend the first step to be:
```yml
    - uses: actions/checkout@v2
      with:
        fetch-depth: 0